vault/src/metrics.rs

```diff
@@ -541,3 +604,91 @@ pub async fn publish_expected_bitcoin_balance<B: BitcoinCoreApi + Clone + Send +
             .set(upperbound as f64 / scaling_factor);
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use runtime::{AccountId, DOT, INTERBTC};
+    use CurrencyId::Token;
+
+    use super::*;
+    fn dummy_redeem_request() -> InterBtcRedeemRequest {
+        InterBtcRedeemRequest {
+            amount_btc: Default::default(),
+            btc_address: Default::default(),
+            btc_height: Default::default(),
+            fee: Default::default(),
+            opentime: Default::default(),
+            period: Default::default(),
+            premium: Default::default(),
+            redeemer: AccountId::new([1u8; 32]),
+            status: RedeemRequestStatus::Pending,
+            transfer_fee_btc: Default::default(),
+            vault: VaultId::new(AccountId::new([1u8; 32]), Token(DOT), Token(INTERBTC)),
+        }
+    }
+
+    #[test]
+    fn test_calculate_remaining_time() {
+        let redeem_period_para_blocks = 5 * 60 * 24;
```

Comment:
Why `5`? It's not clear to me immediately what some of these magic numbers mean. Maybe we can extract some also into constants?